File: app/src/main/java/com/example/sguardaavi/data/AirportManager.java
package com.example.sguardaavi.data;

import com.example.sguardaavi.Utils;
import java.util.List;

public class AirportManager
{
    private static AirportManager instance;

    private List<Airport> airportList;

    private AirportManager(){
        airportList = Utils.Companion.generateAirportList();
    }

    public static AirportManager getInstance(){
        if(instance == null){
            instance = new AirportManager();
        }
        return instance;
    }

    public Airport getAirportByIndex(int index){
        return airportList.get(index);
    }

    public List<Airport> getAirportList()
    {
        return airportList;
    }
}

File: app/src/main/java/com/example/sguardaavi/data/AirportManager.kt
package com.example.sguardaavi.data

import com.example.sguardaavi.Utils.Companion.generateAirportList

class AirportManager private constructor() {
    val airportList: List<Airport>
    fun getAirportByIndex(index: Int): Airport {
        return airportList[index]
    }

    companion object {
        var instance: AirportManager? = null
            get() {
                if (field == null) {
                    field = AirportManager()
                }
                return field
            }
            private set
    }

    init {
        airportList = generateAirportList()
    }
}